File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/gui/hotbar/HotbarController.java
package com.gmail.borlandlp.minigamesdtools.arena.gui.hotbar;

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools;
import com.gmail.borlandlp.minigamesdtools.arena.ArenaComponent;
import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener;
import com.gmail.borlandlp.minigamesdtools.arena.ArenaPhaseComponent;
import com.gmail.borlandlp.minigamesdtools.arena.team.TeamProvider;
import com.gmail.borlandlp.minigamesdtools.creator.DataProvider;
import com.gmail.borlandlp.minigamesdtools.gui.hotbar.Hotbar;
import org.bukkit.entity.Player;
import org.bukkit.event.HandlerList;
import org.bukkit.event.Listener;

public class HotbarController extends ArenaComponent implements ArenaPhaseComponent {
    private String defaultHotbarId;
    private Listener listener;
    private boolean enabled = false;

    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public void setDefaultHotbarId(String defaultH) {
        this.defaultHotbarId = defaultH;
    }

    public String getDefaultHotbarId() {
        return this.defaultHotbarId;
    }

    public Hotbar buildDefaultHotbarFor(Player player) throws Exception {
        String hotbarID = this.getDefaultHotbarId();
        return MinigamesDTools.Companion.getInstance().getHotbarCreatorHub().createHotbar(hotbarID, new DataProvider() {{
            this.set("player", player);
        }});
    }

    @Override
    public void onInit() {
        if(!this.enabled) {
            return;
        }

        this.listener = new HotbarListener(this);
        MinigamesDTools.Companion.getInstance().getServer().getPluginManager().registerEvents(this.listener, MinigamesDTools.Companion.getInstance());
        try {
            this.getArena().getEventAnnouncer().register((ArenaEventListener) this.listener);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public void beforeGameStarting() {
        if(!this.enabled) {
            return;
        }

        for (TeamProvider team : this.getArena().getTeamController().getTeams()) {
            for (Player player : team.getPlayers()) {
                if(player == null) {
                    continue;
                }

                try {
                    MinigamesDTools.Companion.getInstance().getHotbarAPI().bindHotbar(this.buildDefaultHotbarFor(player), player);
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        }
    }

    @Override
    public void gameEnded() {
        if(!this.enabled) {
            return;
        }

        for (TeamProvider team : this.getArena().getTeamController().getTeams()) {
            for (Player player : team.getPlayers()) {
                if(player != null) {
                    MinigamesDTools.Companion.getInstance().getHotbarAPI().unbindHotbar(player);
                }
            }
        }
        HandlerList.unregisterAll(this.listener);
        this.getArena().getEventAnnouncer().unregister((ArenaEventListener) this.listener);
    }

    @Override
    public void update() {

    }

    @Override
    public void beforeRoundStarting() {

    }

    @Override
    public void onRoundEnd() {

    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/gui/hotbar/HotbarController.kt
package com.gmail.borlandlp.minigamesdtools.arena.gui.hotbar

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools.Companion.instance
import com.gmail.borlandlp.minigamesdtools.arena.ArenaComponent
import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener
import com.gmail.borlandlp.minigamesdtools.arena.ArenaPhaseComponent
import com.gmail.borlandlp.minigamesdtools.creator.DataProvider
import com.gmail.borlandlp.minigamesdtools.gui.hotbar.Hotbar
import org.bukkit.entity.Player
import org.bukkit.event.HandlerList
import org.bukkit.event.Listener

class HotbarController : ArenaComponent(), ArenaPhaseComponent {
    var defaultHotbarId: String? = null
    private var listener: Listener? = null
    var isEnabled = false

    @Throws(Exception::class)
    fun buildDefaultHotbarFor(player: Player): Hotbar {
        val hotbarID = defaultHotbarId
        return instance!!.hotbarCreatorHub!!.createHotbar(hotbarID!!, DataProvider().apply {
            this["player"] = player
        })
    }

    override fun onInit() {
        if (!isEnabled) {
            return
        }
        listener = HotbarListener(this)
        instance!!.server.pluginManager.registerEvents(listener, instance)
        try {
            arena!!.eventAnnouncer.register(listener as ArenaEventListener)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun beforeGameStarting() {
        if (!isEnabled) {
            return
        }
        for (team in arena!!.teamController!!.teams) {
            for (player in team.players) {
                try {
                    instance!!.hotbarAPI!!.bindHotbar(buildDefaultHotbarFor(player), player)
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            }
        }
    }

    override fun gameEnded() {
        if (!isEnabled) {
            return
        }
        for (team in arena!!.teamController!!.teams) {
            for (player in team.players) {
                instance!!.hotbarAPI!!.unbindHotbar(player)
            }
        }
        HandlerList.unregisterAll(listener)
        arena!!.eventAnnouncer.unregister(listener as ArenaEventListener)
    }

    override fun update() {}
    override fun beforeRoundStarting() {}
    override fun onRoundEnd() {}
}